pkg/registry/apis/datasource/register.go

```diff
@@ -299,21 +291,30 @@ func (b *DataSourceAPIBuilder) GetOpenAPIDefinitions() openapi.GetOpenAPIDefinit
 	}
 }
 
-func getCorePlugins(cfg *setting.Cfg) ([]plugins.JSONData, error) {
-	coreDataSourcesPath := filepath.Join(cfg.StaticRootPath, "app", "plugins", "datasource")
-	coreDataSourcesSrc := sources.NewLocalSource(
-		plugins.ClassCore,
-		[]string{coreDataSourcesPath},
-	)
+func getDatasourcePlugins(pluginSources sources.Registry) ([]plugins.JSONData, error) {
+	var pluginJSONs []plugins.JSONData
 
-	res, err := coreDataSourcesSrc.Discover(context.Background())
-	if err != nil {
-		return nil, errors.New("failed to load core data source plugins")
-	}
+	// It's possible that the same plugin + version combination will be
+	// in different sources. Registering the same plugin + version combination
+	// twice in the API is Probably A Bad Thing, so this map keeps track of
+	// unique plugin + version combinations.
+	var uniquePluginVersions = map[string]bool{}
 
-	pluginJSONs := make([]plugins.JSONData, 0, len(res))
-	for _, p := range res {
-		pluginJSONs = append(pluginJSONs, p.Primary.JSONData)
+	for _, pluginSource := range pluginSources.List(context.Background()) {
+		res, err := pluginSource.Discover(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		for _, p := range res {
+			if p.Primary.JSONData.Type == plugins.TypeDataSource {
+				pluginIdVersion := p.Primary.JSONData.ID + p.Primary.JSONData.Info.Version
```

Comment:
Ah my bad I just realized we probably only care about the same plugin ID as we're probably registering each API with $pluginID @ `v0alpha1`, right? 

Also no harm in adding a log to the continue in case we do find any dupes!